### PR TITLE
Add Dockerfile, CI settings, and Helm template for index-manager

### PR DESCRIPTION
**Pull request by rinx**
### Description:

I've just added Dockerfile, CI settings, and Helm template for index-manager.
Also i've added implementation related to `auto_index_length` option.

### Related Issue:

Nothing.

### How Has This Been Tested?:

Not yet.

### Environment:

- Golang Version: 1.13.8
- Docker Version: 19.03.5
- Kubernetes Version: 1.16.3
- NGT Version: 1.8.4

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [X] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [X] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [X] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensure all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: vald/values.yaml [hunk 1]
```diff
@@ -653,7 +653,7 @@ indexManager:
   revisionHistoryLimit: 2
   terminationGracePeriodSeconds: 30
```
Comment (rinx) [inline vald/values.yaml]: badge
@kpango Does this component need HPA?

Comment (kpango) [reply vald/values.yaml]: We don't need HPA about this component

Comment (rinx) [reply vald/values.yaml]: Okay. Thanks.

File: Makefile.d/k8s.mk [hunk 2]
```diff
@@ -39,6 +39,9 @@ k8s/manifest/update: \
 	    --set initializer.mysql.secret.enabled=true \
 	    --set initializer.redis.secret.enabled=true \
 	    --set initializer.cassandra.secret.enabled=true \
+	    --set agent.minReplicas=5 \
+	    --set gateway.gateway_config.discoverer.agent_client.connection_pool=10 \
+	    --set indexManager.indexer.discoverer.agent_client.connection_pool=10 \
```
Comment (kpango) [inline Makefile.d/k8s.mk]: for index manager we don't need performance about indexing please set it to 2
and please take care about keep alive setting, indexin needs long time connection

File: pkg/manager/index/service/option.go [hunk 3]
```diff
@@ -34,6 +34,8 @@ var (
 		WithIndexingConcurrency(1),
 		WithIndexingDuration("1m"),
 		WithIndexingDurationLimit("30m"),
+		WithMinUncommitted(100),
+		WithAgentNamespace("default"),
```
Comment (kpango) [inline pkg/manager/index/service/option.go]: Also I forgot implementation about node selecting, could you add it? same as namespace

File: vald/values.yaml [hunk 4]
```diff
@@ -680,6 +680,16 @@ indexManager:
...
+  indexer:
+    concurrency: 1
+    auto_index_duration_limit: 30m
+    auto_index_check_duration: 1m
+    auto_index_length: 100
+    discoverer:
+      duration: "2s"
+      discover_client: {}
+      agent_client:
+        connection_pool: 100
```
Comment (kpango) [inline vald/values.yaml]: may be 3~5 is ok

File: Makefile.d/k8s.mk (hunk 2 above)
Comment (rinx) [reply Makefile.d/k8s.mk]: how about to set `indexManager.indexer.discoverer.agent_client.dial_option.tcp.dialer.keep_alive=5m` ?

Comment (kpango) [reply Makefile.d/k8s.mk]: 5minute is not enough how about set it to 15m for maximum but it's upto minUncommitted index length :angel:

**Comment by rinx** [comment]
@kpango I've just fixed. please check them. :pray:

File: vald/values.yaml [hunk 5]
```diff
@@ -232,6 +232,15 @@ gateway:
       target: agent
       image: busybox
       sleepDuration: 2
+  env:
+    - name: MY_NODE_NAME
+      valueFrom:
+        fieldRef:
+          fieldPath: spec.nodeName
```
Comment (kpango) [inline vald/values.yaml]: setting node name is optional for gateway, if we define nodename for discovery we should set component kind as DaemonSet, but gateway needs performance we can skip this setting.
node name config for gateway will be used in node level gateway

File: vald/values.yaml [hunk 6]
```diff
@@ -662,6 +672,15 @@ indexManager:
   rollingUpdate:
     maxSurge: 25%
     maxUnavailable: 25%
+  env:
+    - name: MY_NODE_NAME
+      valueFrom:
+        fieldRef:
+          fieldPath: spec.nodeName
```
Comment (kpango) [inline vald/values.yaml]: similar things as gateway

File: vald/values.yaml (hunk 5 above)
Comment (rinx) [reply vald/values.yaml]: I've just commented out these settings. thanks :+1:

**Comment by kpango** [comment]
LGTM
/rebase
/format
/approve

**Pull request merged**